### PR TITLE
Feature/perl symbols

### PR DESCRIPTION
**Pull request by idodeclare**
Hello,

Please consider for merging this patch with another set of Perl improvements:

* improvements to parsing of quote-like operators, including a fix to include symbols in quoting characters, a fix to recognize spaces around braced identifiers, etc.
* extract Perl symbols using the equivalent parsing as done for xref, and add a test of extracted Perl symbols.
* add Perl pragmas as keywords
* recognize Perl special identifiers, and format as keywords
* add PLX as a Perl suffix, to align with ctags perl.c
* improve POD recognition
* clean up to use existing yypush(), yypop(); and add yyjump()

I added test lines for all the improvements and fixes. Now that feature/perl_xref_tests is merged, I can raise this PR.

Thank you.

### CODE REVIEW

**Comment by idodeclare** [comment]
Oh, I should point out that to rationalize PerlXref.lex and PerlSymbolTokenizer.lex where the latter was case-sensitive and the former was not and also in order to improve POD handling, I revised PerlXref.lex to be case-sensitive in 77053af1d7. This required modifying just the POD-related patterns and the {FileExt} pattern.

**Comment by tarzanek** [comment]
I don't know why the build suddenly failed with javadoc errors, I hoped travis builds are ignoring the javadoc errors, seems like we need to update pom.xml to get a fix for the plugin for javadoc

File: test/org/opensolaris/opengrok/analysis/perl/sample.pl [hunk 1]
```diff
@@ -1,280 +0,0 @@
-#!/usr/bin/perl
```
Comment (tarzanek) [inline test/org/opensolaris/opengrok/analysis/perl/sample.pl]: symlink ... this will make development problematic on windows ...

**Comment by tarzanek** [comment]
lgtm, let's merge ...
@vladak @tulinkry  any comments ?

**Comment by idodeclare** [comment]
I've been extending this pull request commit-by-commit. Each commit has stood on its own as a working version (along with its predecessors) and could have been merged at any time.

I finished rationalizing the differences between PerlXref and PerlSymbolTokenizer and pushed up a last modification for them to %include the same PerlProductions.lex.

**Comment by tarzanek** [comment]
great, includes is something I've been longing to get to jflex files for a long time!
let me review

**Comment by tarzanek** [comment]
fwiw the problem with includes was that they didn't really accept relative paths, not sure if this improved in last jflex or not

**Comment by tarzanek** [comment]
but if it works, then I guess we can merge ...

**Comment by idodeclare** [comment]
@tarzanek , in debugging the %include syntax, where the 1.6.1 manual reads to use quotation marks but where that doesn't actually work, I did try at one point an absolute path with and without quotation marks without success. Possibly only local %include works.

**Comment by tarzanek** [comment]
yeah, local works, I tried to abstract stuff from all lex files to top dir and that didn't work

**Comment by tarzanek** [comment]
ok, let's merge this

**Comment by idodeclare** [comment]
@tarzanek , perhaps it would do to revise the &lt;run-jflex&gt; and &lt;run-jflex-and-disable-buffer-expansion&gt; targets to copy(if-updated) the source .lex files to an intermediate directory, including shared includes?

**Comment by tarzanek** [comment]
hmm , that could be a good idea and we could sacrifice space/time for easier maintainance of lex files ... and especially keeping fixes and improvements in sync

**Pull request merged**